packages/core/src/external/athenahealth/index.ts

```diff
@@ -181,6 +185,57 @@ class AthenaHealthApi {
     }
   }
 
+  async getPatientViaSearch({
+    cxId,
+    patientId,
+  }: {
+    cxId: string;
+    patientId: string;
+  }): Promise<PatientResource | undefined> {
+    const { log, debug } = out(`AthenaHealth search - AH patientId ${patientId}`);
+    try {
+      const data = {
+        _id: this.createPatientId(patientId),
+        "ah-practice": this.createPracticetId(this.practiceId),
+      };
+      const patienSearchtUrl = "/Patient/_search";
+      const response = await this.handleAxiosRequest(() =>
+        this.axiosInstanceFhirApi.post(patienSearchtUrl, this.createDataParams(data))
+      );
+      if (!response.data) throw new Error(`No body returned from ${patienSearchtUrl}`);
+      debug(`${patienSearchtUrl} resp: ${JSON.stringify(response.data)}`);
+      if (responsesBucket) {
+        const filePath = createHivePartitionFilePath({
+          cxId,
+          patientId,
+          date: new Date(),
+        });
+        const key = `athenahealth/patient-search/${filePath}/${uuidv7()}.json`;
+        await this.s3Utils.uploadFile({
+          bucket: responsesBucket,
+          key,
+          file: Buffer.from(JSON.stringify(response.data), "utf8"),
+          contentType: "application/json",
+        });
+      }
```

Comment:
Can't this be executed in parallel? Not sure if this is running on a lambda env or not

Reply:
How do you mean? We have to wait for the response from the API.

Reply:
is the file used for processing the request/feature? I thought it was just storing the response/request on S3 for debugging purposes.

Reply:
It's just for storing -- just not sure what "in parallel" you mean? Do you mean remove the await and do it async?

Reply:
Will address in follow up utilities PR.

Reply:
https://github.com/metriport/metriport/pull/2835/files